scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/goals/CardinalityGoal.java

```diff
@@ -120,19 +137,31 @@ protected CardinalityGoal fill(CardinalityGoal goal) {
       if (occurrenceRange != null) {
         goal.occurrenceRange = occurrenceRange;
       }
+      if(isUnsatisfiableDurationType()){
+        throw new IllegalArgumentException("Goal is incorrectly parametrized: activity creation template has zero-duration while duration objective is non-zero");
+      }
       if(isUnsatisfiable()){
-        throw new IllegalArgumentException("Goal is incorrectly parametrized and therefore unsatisfiable : minimum duration x minimum occurence > maximum duration");
+        throw new IllegalArgumentException("Goal is incorrectly parametrized: minimum duration x minimum occurence > maximum duration");
       }
       return goal;
     }
 
+    public boolean isUnsatisfiableDurationType(){
+      return (thereExists.getDurationRange() != null &&
+              thereExists.getDurationRange().isSingleton() &&
+              thereExists.getDurationRange().start.isZero() &&
+              (thereExists.getType().getDurationType() instanceof DurationType.Controllable) &&
```

Comment:
I'm curious why the duration type is needed for `isUnsatisfiableDuraitonType`? I think I just don't quite understand this method.

Reply:
You are right, it is not required, I'll remove it.